frontend/src/app/shared/custom/scoring/scoring.component.ts

```diff
@@ -105,19 +105,46 @@ export class ScoringComponent implements OnInit, AfterViewInit, OnChanges {
       const keyResult = this.keyResult as KeyResultMetricMin;
       const lastCheckIn = keyResult.lastCheckIn?.value!;
 
-      if (lastCheckIn < keyResult.commitValue) {
+      if (keyResult.baseline === keyResult.stretchGoal) {
+        if (lastCheckIn >= keyResult.baseline) {
+          this.stretched = true;
+          this.failPercent = 100;
+          this.commitPercent = 100;
+          this.targetPercent = 100;
+        } else {
+          this.stretched = false;
+          this.failPercent = 0;
+          this.commitPercent = 0;
+          this.targetPercent = 0;
+        }
+        return;
+      }
+
+      // Decide comparison direction based on goal orientation
+      const increasing = keyResult.baseline < keyResult.stretchGoal;
```

Comment:
You can probably delete the upper if statement if you add a `=` to the `<` here.

```suggestion
      const increasing = keyResult.baseline <= keyResult.stretchGoal;
```